lib/entities/resource-type.ts

```diff
@@ -0,0 +1,140 @@
+import type {
+  BasicMetaSysProps,
+  DefaultElements,
+  GetResourceTypeParams,
+  MakeRequest,
+  SysLink,
+} from '../common-types'
+import { toPlainObject, freezeSys } from 'contentful-sdk-core'
+import copy from 'fast-copy'
+import enhanceWithMethods from '../enhance-with-methods'
+
+export type ResourceTypeProps = {
+  /**
+   * System metadata
+   */
+  sys: Omit<BasicMetaSysProps, 'version'> & {
+    appDefinition: SysLink
+    organization: SysLink
+    resourceProvider: SysLink
+  }
+  /**
+   * Resource Provider name, TODO
```

Comment:
Are the two `TODO` still to be done? What is missing?

Reply:
it's done yes, forgot to remove it